File: front/src/types/Producto.ts
import Receta from "./Receta";
import RubroProducto from "./RubroProducto";

export default interface Producto {
    id: number,
    fechaAlta: Date,
    fechaModificacion: Date,
    fechaBaja: Date,
    denominacion: string,
    descripcion: string,
    tiempoEstimadoCocina: number,
    precioVenta: number,
    costo: number,
    urlImagen: string,
    rubroProducto: RubroProducto,
    receta: Receta
}

File: back/src/main/java/com/utn/elbuensabor/entities/productos/Producto.java
package com.utn.elbuensabor.entities.productos;

import com.utn.elbuensabor.entities.Base;
import jakarta.persistence.*;
import lombok.*;
import org.antlr.v4.runtime.misc.NotNull;
import java.math.BigDecimal;
import java.util.*;

@Entity
@Table(name = "articulo_manufacturado")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Builder
public class Producto extends Base {

    @NotNull
    private String denominacion;

    @NotNull
    @Column(length = 1000)
    private String descripcion;

    @NotNull
    @Column(name = "tiempo_estimado_cocina")
    private Integer tiempoEstimadoCocina;

    @NotNull
    @Column(name = "precio_venta", precision = 10, scale = 2)
    private BigDecimal precioVenta;

    @Column(name = "costo", precision = 10, scale = 2)
    private BigDecimal costo;

    @Column(length = 500, name = "url_imagen")
    private String urlImagen;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "id_rubro_producto")
    private RubroProducto rubroProducto;

    @NotNull
    @OneToMany (mappedBy = "id_producto")
    private List<DetalleReceta> receta = new ArrayList();

}
